packages/connect-node/src/transport.spec.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable @typescript-eslint/no-invalid-void-type */
+import { Int32Value, StringValue, MethodKind } from "@bufbuild/protobuf";
+import { useNodeServer } from "./use-node-server-helper.spec.js";
+import * as http2 from "node:http2";
+import { connectNodeAdapter } from "./connect-node-adapter.js";
+import { createPromiseClient } from "@connectrpc/connect";
+import { createTransport as createGrpcTransport } from "@connectrpc/connect/protocol-grpc";
+import { createTransport as createGrpcWebTransport } from "@connectrpc/connect/protocol-grpc-web";
+import { validateNodeTransportOptions } from "./node-transport-options.js";
+
+const TestService = {
+  typeName: "TestService",
+  methods: {
+    server: {
+      name: "Server",
+      I: Int32Value,
+      O: StringValue,
+      kind: MethodKind.ServerStreaming,
+    },
+  },
+} as const;
+
+describe("Calls should fail with code internal on RST_STREAM no_error before trailers are received", function () {
```

Comment:
Could we possibly add this as a custom connectconformance test with the flag `--test-file`?

Reply:
There is no way to trigger an `RST_STREAM(0x00)` using a test file. 

We can indirectly trigger it by providing a raw http response that ends the response before request is finished, but that also needs code changes in the conformance runner to allow that case. It [drains the requests](https://github.com/connectrpc/conformance/blob/368943e8bfff29a06fda135b8680433516dbcde5/internal/app/referenceserver/raw_response.go#L220-L228) before sending the raw response.